client/ayon_core/hosts/substancepainter/api/lib.py

```diff
@@ -643,3 +640,86 @@ def _setup_prompt():
         return
 
     return project_mesh
+
+
+def convert_substance_object_to_python(subst_proj_option="default"):
+    """Function to convert substance C++ objects to python instance.
+    It is made to avoid any possible ValueError when C++ objects casting
+    as python instance.
+
+    Args:
+        subst_proj_option (str, optional): Substance project option.
+            Defaults to "default".
+
+    Raises:
+        ValueError: Raise Error when unsupported Substance
+            Project was detected
+
+    Returns:
+        python instance: converted python instance of the C++ objects.
+    """
+    if subst_proj_option == "default":
+        return substance_painter.project.ProjectWorkflow.Default
+    elif subst_proj_option == "uvTile":
+        return substance_painter.project.ProjectWorkflow.UVTile
+    elif subst_proj_option == "textureSetPerUVTile":
+        return substance_painter.project.ProjectWorkflow.TextureSetPerUVTile
+    elif subst_proj_option == "PerFragment":
+        return substance_painter.project.TangentSpace.PerFragment
+    elif subst_proj_option == "PerVertex":
+        return substance_painter.project.TangentSpace.PerVertex
+    elif subst_proj_option == "DirectX":
+        return substance_painter.project.NormalMapFormat.DirectX
+    elif subst_proj_option == "OpenGL":
+        return substance_painter.project.NormalMapFormat.OpenGL
+    else:
+        raise ValueError(
+            f"Unsupported Substance Objects: {subst_proj_option}")
+
+
+def parse_substance_attributes_setting(template_name, project_templates):
+    """Function to parse the dictionary from the AYON setting to be used
+    as the attributes for Substance Project Creation
+
+    Args:
+        template_name (str): name of the template from the setting
+        project_templates (dict): project template data from the setting
+
+    Returns:
+        dict: data to be used as attributes for Substance Project Creation
+    """
+    attributes_data = {}
+    for template in project_templates:
+        if template["name"] == template_name:
+            attributes_data.update(template)
+    attributes_data["normal_map_format"] = convert_substance_object_to_python(
+        subst_proj_option=attributes_data["normal_map_format"])
+    attributes_data["project_workflow"] = convert_substance_object_to_python(
+        subst_proj_option=attributes_data["project_workflow"])
+    attributes_data["tangent_space_mode"] = convert_substance_object_to_python(
+        subst_proj_option=attributes_data["tangent_space_mode"])
+    attributes_data.pop("name")
+    attributes_data.pop("preserve_strokes")
+    return attributes_data
```

Comment:
This feels very specific to some part of the settings or logic, don't see why these need to be in `lib`. They don't seem generic enough.